bin/dfx-sns-config-random

```diff
@@ -21,6 +36,15 @@ PRINCIPAL="$(dfx identity get-principal)"
 # For original cotrollers, use:
 # dfx canister info smiley_dapp --network "$DFX_NETWORK" | perl -e 'while($_ = <>){ if (s/^Controllers: *//){s/ +/, /g;print $_} }'
 
-sed "s/TOKEN_NAME/$TOKEN_NAME/g;s/TOKEN_SYMBOL/$TOKEN_SYMBOL/g;s/PRINCIPAL/$PRINCIPAL/g" "$(dirname "$(realpath "$0")")"/sns_init.yml >sns.yml
+SNS_INIT_PATH="$(dirname "$(realpath "$0")")/sns_init.yml"
+SED_COMMANDS="s/TOKEN_NAME/$TOKEN_NAME/g;
+              s/TOKEN_SYMBOL/$TOKEN_SYMBOL/g;
+              s/PRINCIPAL/$PRINCIPAL/g"
+
+if [ "${CONFIRMATION_TEXT:-}" ]; then
+  SED_COMMANDS="$SED_COMMANDS;s/# confirmation_text: \"CONFIRMATION_TEXT\"/confirmation_text: \"${CONFIRMATION_TEXT}\"/"
```

Comment:
Does this add the "confirmation_text" field in sns_yaml.init and "uncomments it?

Reply:
Yes, but it does it by taking all of
```
# confirmation_text: "CONFIRMATION_TEXT"
```
and replacing it with
```
confirmation_text: "${CONFIRMATION_TEXT}"
```
It doesn't separately remove `#` and fill in the placeholder or anything like that.

In general `sed 's/foo/bar/'` replaces foo with bar.

I'll add a comment.

Reply:
Thanks for the explanation!

Reply:
Thanks, I agree it's fine to keep it simple as it is.